autometa/common/external/diamond.py

```diff
@@ -29,146 +29,301 @@
+    -------
+    str
+        Path to Diamond database
+
+    Raises
+    ------
+    CalledProcessError
+        Failed to create Diamond database
+    """
+    cmd = f"diamond makedb --in {fasta} --db {database} -p {nproc}"
+    logger.debug(f"{cmd}")
+    proc = subprocess.run(
+        cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, text=True
+    )
+    try:
+        proc.check_returncode()
+    except subprocess.CalledProcessError as err:
+        raise err
     return database
 
-def blast(fasta, database, outfpath, blast_type='blastp', evalue=float('1e-5'),
-    maxtargetseqs=200, cpus=0, tmpdir=os.curdir, force=False, verbose=False):
-    """Performs diamond blastp search using fasta against diamond formatted database
+
+def blast(
+    fasta,
+    database,
+    outfpath,
+    blast_type="blastp",
+    evalue=float("1e-5"),
+    maxtargetseqs=200,
+    nproc=mp.cpu_count(),
+    tmpdir=os.curdir,
+    force=False,
+    verbose=False,
+):
+    """
+    Performs diamond blastp search using query sequence against diamond formatted database
 
     Parameters
     ----------
     fasta : str
-        </path/to/fasta/file>. May be amino acid or nucleotide sequences
+        Path to fasta file having the query sequence. May be amino acid or nucleotide sequences
+        '</path/to/fasta/file>' 
     database : str
-        </path/to/diamond/formatted/database>.
+        Path to diamond formatted database
+        '</path/to/diamond/formatted/database>'
     outfpath : str
-        </path/to/output/file>.
-    blast_type : str
-        blastp if fasta consists of amino acids or blastx if nucleotides
-    evalue : float
-        cutoff e-value to count hit as significant (the default is float('1e-5')).
-    maxtargetseqs : int
-        max number of target sequences to retrieve per query by diamond (the default is 200).
-    cpus : int
-        number of cpus to use (the default is 1).
-    tmpdir : type
-        </path/to/temporary/directory> (the default is os.curdir).
-    force : boolean
-        overwrite existing diamond results `force` (the default is False).
-    verbose : boolean
-        log progress to terminal `verbose` (the default is False).
+        Path to output file
+        '</path/to/output/file>'
+    blast_type : str, optional
+        blastp to align protein query sequences against a protein reference database,
+        blastx to align translated DNA query sequences against a protein reference database, by default 'blastp'
+    evalue : float, optional
+        cutoff e-value to count hit as significant,, by default float('1e-5')
+    maxtargetseqs : int, optional
+        max number of target sequences to retrieve per query by diamond, by default 200
+    nproc : int, optional
+        Number of processors to be used, by default uses all the processors of the system
+    tmpdir : str, optional
+        Path to temporary directory
+        '</path/to/temporary/directory>' by default os.curdir
```

Comment:
I'm not sure having both of these lines is helpful.

```suggestion
        Path to temporary directory, by default os.curdir
```

The default should probably _not_ be the current directory, but rather an actual temp directory. You could probably implement this using the [tempfile](https://docs.python.org/3/library/tempfile.html) module